### PR TITLE
[BugFix] Set strict=False in tensordict.select() calls for objective classes

### PR DESCRIPTION
**Pull request by albertbou92**
## Description

This PR proposes to set `strict=False` for the `tensordict.select()` calls in the objective classes.

The logic behind this is that in some situations the Actors and Value networks should still be able to run through the losses without some of the in_keys. 

Examples are: 
1. RNN without the `recurrent_state` key can still be called when the `is_init` is True.
2. Transformer networks may need a mask during inference but not during training.
3. More generally, users could make custom TensorDict modules that have optional parameters. 

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/objectives/a2c.py [hunk 1]
```diff
@@ -411,7 +411,9 @@ def _log_probs(
             raise RuntimeError(
                 f"tensordict stored {self.tensor_keys.action} require grad."
             )
-        tensordict_clone = tensordict.select(*self.actor_network.in_keys).clone()
```
Comment (vmoens) [inline torchrl/objectives/a2c.py:414]: is it required to change it in the tests too?

File: torchrl/objectives/a2c.py [hunk 2]
```diff
@@ -425,7 +427,9 @@ def loss_critic(self, tensordict: TensorDictBase) -> torch.Tensor:
             # TODO: if the advantage is gathered by forward, this introduces an
             # overhead that we could easily reduce.
             target_return = tensordict.get(self.tensor_keys.value_target)
-            tensordict_select = tensordict.select(*self.critic_network.in_keys)
```
Comment (vmoens) [inline torchrl/objectives/a2c.py:428]: ditto

**Review by vmoens — APPROVED** [review]
LGTM just curious why we needed to adapt the tests too
